test/hotspot/jtreg/compiler/codegen/BMI2.java

```diff
@@ -0,0 +1,449 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Support BMI2 instructions on x86/x64
+ *
+ * @run main/othervm -Xbatch -XX:-TieredCompilation -XX:+UnlockDiagnosticVMOptions -XX:+AbortVMOnCompilationFailure
+ *      -XX:CompileCommand=compileonly,compiler.codegen.BMI2$BMITests::*
+ *      compiler.codegen.BMI2
+ */
+
+package compiler.codegen;
+
+public class BMI2 {
+    private final static int ITERATIONS = 30000;
+
+    // match(Set dst (ConvI2L (AndI src1 mask))) // Using 0
+    public static void testZeroBzhiI2L() {
+        int ix = 0;
+        int i = 0;
+
+            long z = BMITests.bzhiI2L(ix, i);
+
+            for (int i2 = 0; i2 < ITERATIONS; i2++) {
+                long ii = BMITests.bzhiI2L(ix, i);
+                if (ii != z) {
+                    throw new Error(returnBzhiI2LErrMessage (z, ii));
+                }
+            }
+
+            i = 1;
+
+            z = BMITests.bzhiI2L(ix, i);
+
+            for (int i2 = 0; i2 < ITERATIONS; i2++) {
```

Comment:
Why are these manually unrolled? They are the same but with different index?

Reply:
I wasn't able to get the above to C2 compile separately. I wrote essentially the same but could not verify the compilation. I then unrolled from the for loop and could verify the bzhi instructions were generated. Repetition felt like a reasonable compromise for completeness. I'll take another look and see if I can verify in a for loop. Perhaps one of the options I was using prevented it from working.

Reply:
Rolling this into for loops appears to work. I'm guessing something else was at issue before when I couldn't verify bzhiq instructions in C2 compilation of bzhiI2L(). Pushed a patch with the change.